lib/AutoPodArgsPass.cpp

```diff
@@ -91,6 +126,7 @@ void clspv::AutoPodArgsPass::runOnFunction(Function &F) {
   // 4. No arrays.
   // 5. If 16-bit types are used, 16-bit push constants are supported.
   // 6. If 8-bit types are used, 8-bit push constants are supported.
+  // 7. Not to have a image channel getter function call.
```

Comment:
The documentation about this function support should be updated. That documentation should also describe the current limitations. For example, clspv does not account for the right push constant size yet when channel order or channel formats are used.

Reply:
I am not sure to understand what documentation I should update, could you precise please?
Also, I don't get where clspv tells the right push constant size?

Reply:
docs/OpenCLOnVulkan.md describes the state of support that clspv has for opencl.

There is a section that lists get_image_channel_order() and get_image_channel_data_type() as unsupported. 

There is an option in clspv, `-max-pushconstant-size`, that specifies the max size of push constants. It defaults to the minimum requirement in Vulkan, but clvk will pass the right value for the implementation. Clspv uses this option to determine if POD args will fit in the push constants (also accounting for the other members, e.g. region offset). If they can't fit clspv has to choose another implementation (UBO). The size of the image metadata is unknown at this point in the flow and there is currently no fall back implementation for image metadata. If the POD args (plus earlier elements) fit in the limit, but the image metadata pushes that size over the limit the SPIR-V will be invalid for the implementation. This will lead to incorrect results or a crash in clvk. I'm ok with limiting the initial implementation, but we need to advertise that limitation until the fallback is implemented.

Reply:
I have updated the documentation